python/python/bystro/prs/prs_listener.py

```diff
@@ -0,0 +1,150 @@
+"""Provide a worker for PRS calculation."""
+
+import argparse
+import logging
+from pathlib import Path
+
+from bystro.beanstalkd.messages import BaseMessage, CompletedJobMessage, SubmittedJobMessage
+from bystro.beanstalkd.worker import ProgressPublisher, QueueConf, get_progress_reporter, listen
+from bystro.utils.timer import Timer
+
+# TODO: Define PRS calculation and preprocess covariate, association files
+from bystro.prs.prs_types import PRSResponse, PRSSubmission
+from bystro.prs.calculate_prs import (
+    _load_dosage_matrix,
+    _load_association_scores,
+    _load_covariates,
+    calculate_prs_scores,
+)
+
+# TODO: Define how we load vcf-like genotype matrix from annotation,
+# then generalize this process for other analyses
+from bystro.prs.utils import load_vcf_from_annotation
+
+# TODO: Implement function to preprocess genotype data from annotation
+# into format suitable for PRS calculation
+from bystro.prs.utils import preprocess_vcf_for_prs
+
+from ruamel.yaml import YAML
+
+logging.basicConfig(
+    filename="prs_worker.log",
+    level=logging.DEBUG,
+    format="%(asctime)s.%(msecs)03d %(levelname)s %(module)s - %(funcName)s: %(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+logger = logging.getLogger()
+
+PRS_TUBE = "prs_analysis"
+
+
+class PRSJobData(BaseMessage, frozen=True):
+    """Wrap a PRSSubmission in a BaseMessage for beanstalkd."""
+
+    prs_submission: PRSSubmission
```

Comment:
Instead of doing this, more idiomatic use of Struct (and similarly for other slot classes, like Dataclass):

```python
class PRSJobData(BaseMessage, frozen=True):
   dosage_matrix_path: Path
   association_score_filepath: Path
   covariate_filepath: Path
```

Reply:
If you want to keep the structure you had, with a submission_id property and a prs_submission property that contains the input data, I would just make PRSSubmission a Struct, in which case PRSJobData doesn't change, remaining as:
```python
class PRSJobData(BaseMessage, frozen=True):
    """Wrap a PRSSubmission in a BaseMessage for beanstalkd."""

    prs_submission: PRSSubmission
```
but PRSSubmission becomes:

```python
import Struct form msgspec

class PRSSubmission(Struct, frozen=True):
   dosage_matrix_path: Path
   association_score_filepath: Path
   covariate_filepath: Path
```